src/TimingData.cpp

```diff
@@ -267,15 +266,15 @@ void TimingData::ShiftRange(int start_row, int end_row,
 			int last_row= std::max(end_row, end_row + shift_amount);
 			int first_affected= GetSegmentIndexAtRow(seg_type, first_row);
 			int last_affected= GetSegmentIndexAtRow(seg_type, last_row);
-			if(first_affected == INVALID_INDEX)
+			if(first_affected == INVALID_INDEX || last_affected == INVALID_INDEX)
 			{
 				continue;
 			}
 			// Prance through the affected area twice.  The first time, changing
 			// the rows of the segments, the second time removing segments that
 			// have been run over by a segment being moved.  Attempts to combine
 			// both operations into a single loop were error prone. -Kyz
-			for(size_t i= first_affected; i <= last_affected && i < segs.size(); ++i)
+			for(size_t i= first_affected; last_affected >= 0 && i <= static_cast<uint32_t>(last_affected) && i < segs.size(); ++i)
```

Comment:
I think we can just do `last_affected < 0` to the if condition above and remove the these added conditions.

Reply:
I guess the `if` condition above is essentially already doing that check, so I'll just remove the non-negative checks here.